crates/game/game-lib/src/lib.rs

```diff
@@ -1,5 +1,72 @@
+use core_crnn::thinking_layer::ThinkingLayer;
 use std::time::Duration;
 
 pub trait Game {
+    fn from_model(model: ThinkingLayer) -> Self;
+
+    fn input_nodes() -> usize;
+    fn output_nodes() -> usize;
+
+    fn run(&mut self, game_settings: GameSettings) -> f32 {
+        let tick_count =
+            game_settings.duration.as_millis() / game_settings.tick_duration.as_millis();
+
+        for _ in 0..game_settings.pre_ticks {
+            self.tick_model()
+        }
```

Comment:
How will this help Impact training / performance of the model. Do we input data during the preticks?

Reply:
> Do we input data during the preticks?
would say so.

my idea was that the model might need a few ticks to propagate all the information.
this might not be important for pong.
But for a car e.g. it would be pretty bad if it starts by steering full left to then slowly settle in.

Reply:
Ok yes makes sense. Should probably be a function of the crnn. It knows how many ticks it needs at least for input to reach output. But this is for the future.

Reply:
each Tinking layer add an at least 1 tick delay

Reply:
will be done in a future PR - #6